pkg/cmd/pr/merge/merge.go

```diff
@@ -148,51 +162,100 @@ func mergeRun(opts *MergeOptions) error {
 
 	deleteBranch := opts.DeleteBranch
 	crossRepoPR := pr.HeadRepositoryOwner.Login != baseRepo.RepoOwner()
-	isTerminal := opts.IO.IsStdoutTTY()
 
 	isPRAlreadyMerged := pr.State == "MERGED"
 	if !isPRAlreadyMerged {
-		mergeMethod := opts.MergeMethod
+		payload := mergePayload{
+			repo:          baseRepo,
+			pullRequestID: pr.ID,
+			method:        opts.MergeMethod,
+			auto:          opts.AutoMerge,
+			commitBody:    opts.Body,
+		}
+		if opts.Body != "" {
+			payload.setCommitBody = true
+		}
 
 		if opts.InteractiveMode {
 			r, err := api.GitHubRepo(apiClient, baseRepo)
 			if err != nil {
 				return err
 			}
-			mergeMethod, err = mergeMethodSurvey(r)
+			payload.method, err = mergeMethodSurvey(r)
 			if err != nil {
 				return err
 			}
 			deleteBranch, err = deleteBranchSurvey(opts, crossRepoPR)
 			if err != nil {
 				return err
 			}
-			confirm, err := confirmSurvey()
+
+			allowEditMsg := payload.method != PullRequestMergeMethodRebase
+
+			action, err := confirmSurvey(allowEditMsg)
 			if err != nil {
-				return err
+				return fmt.Errorf("unable to confirm: %w", err)
+			}
+
+			if action == shared.EditCommitMessageAction {
+				var editorCommand string
+				editorCommand, err = cmdutil.DetermineEditor(opts.Config)
+				if err != nil {
+					return err
+				}
+
+				if payload.commitBody == "" {
+					if payload.method == PullRequestMergeMethodMerge {
+						payload.commitBody = pr.Title
+					} else {
+						payload.commitBody = pr.ViewerMergeBodyText
+					}
+				}
+
+				payload.commitBody, err = commitMsgSurvey(payload.commitBody, editorCommand)
+				if err != nil {
+					return err
+				}
+				payload.setCommitBody = true
+
+				action, err = confirmSurvey(false)
+				if err != nil {
+					return fmt.Errorf("unable to confirm: %w", err)
+				}
 			}
-			if !confirm {
+			if action == shared.CancelAction {
 				fmt.Fprintln(opts.IO.ErrOut, "Cancelled.")
 				return cmdutil.SilentError
 			}
 		}
 
-		err = api.PullRequestMerge(apiClient, baseRepo, pr, mergeMethod, opts.Body)
+		err = mergePullRequest(httpClient, payload)
 		if err != nil {
 			return err
 		}
 
 		if isTerminal {
-			action := "Merged"
-			switch mergeMethod {
-			case api.PullRequestMergeMethodRebase:
-				action = "Rebased and merged"
-			case api.PullRequestMergeMethodSquash:
-				action = "Squashed and merged"
+			if payload.auto {
+				method := ""
+				switch payload.method {
+				case PullRequestMergeMethodRebase:
+					method = " via rebase"
+				case PullRequestMergeMethodSquash:
+					method = " via squash"
+				}
+				fmt.Fprintf(opts.IO.ErrOut, "%s Pull request #%d will automatically get merged%s when all requirements are met\n", cs.SuccessIconWithColor(cs.Green), pr.Number, method)
```

Comment:
Just a small wording tweak
```suggestion
				fmt.Fprintf(opts.IO.ErrOut, "%s Pull request #%d will be automatically merged%s when all requirements are met\n", cs.SuccessIconWithColor(cs.Green), pr.Number, method)
```

I don't love "when all requirements are met" but I can't think of anything better either 🤔

Reply:
Just curious if displaying the merge methods here is accurate? From the API request for enabling automerge it doesn't seem like we specify how the PR is to be merged.